### ISSUE TITLE
Trouble finding trend, and seasonality, and cycles with SARIMA

### ISSUE THREAD

**Issue by samuel-1985** [statsmodels/statsmodels#4688]
Hi
I am a newbie to this so I already apologize if my questions sound too simple.

I have a daily time series and want to do daily forecasting that involves irregular seasonal effects or causal variables like holidays. the data should have messy seasonal factors such as day of the month etc., holiday effects, sometimes moving holiday effects (Example: Easter). How can I factor in all these components in SARIMA? Also, how can I not only capture the effect on the holiday but also the effect surrounding that holiday with SARIMA? For instance, people don't shop on New Years day. All the shopping starts at least 2 weeks prior to new years day and ramps down after New Years day. The weeks surrounding the Holiday will be different from a typical week say you observe in a typical week February.
Moreover, if my data is daily data, how can I capture any possible seasonality including weekly, monthly, and annually? Do I do it by multiplicative seasonal components? If yes, how should the command look like?

Please advise how my SARIMA function, and its arguments should look like.
Regards
Samuel

**Comment by ChadFulton** [comment]
I think this is a very broad question about time series modeling generally, so it will be pretty difficult to give you a general answer.

Often times people handle many of the effects you're interested in with dummy-type exogenous variables, which our SARIMAX class can handle via the `exog` parameter, although it does not generate any of them for you. For useful examples of what types of effects you might include and how to formulate them, you might find the Census Bureau's X13 documentation useful, e.g. Table 4.1 (starts on page 31) of https://www.census.gov/ts/x13as/docX13AS.pdf.

SARIMAX can incorporate seasonality of varying lengths using the `seasonal_order` parameter, but if your data is daily, then it may be possible to include annual seasonal effects, but it will be computationally demanding.

**Comment by samuel-1985** [comment]
@ChadFulton Thanks so much for the reply. I was particularly pointing at the how to type the SARIMA command. I thought that it would be something like: SARIMA(order=(6,1,4), seasonal order=(2,1,1,7)x(2,1,1,30)x(2,1,1,365)) to basically account for weekly, monthly, and annually seasonalities in a multiplicative way. Is that correct? if not, how to do so with dummy variables? Can you please give some insights with the code? 
And my last question would be why to use exogenous variables while all I have is a date and corresponding quantity?

**Comment by REspanha** [comment]
Hi Samuel,

I was having the same problem with ARIMA. In my case I have daily cycles with weekly seasons too. Imagine a number of cars per hour dataset. During the week I have some daily pattern but in the weekend it drops. I was looking for a model to capture multi seasonality and I found TBATS. I gave it a try but no clear success. I ended up using SARIMA with order (5,1,0) and seasonal order (0,1,0,24*7). The 5,1,0 was able to catch the daily cycles by removing the cycle with differencing (1)and the seasonal param m=24*7 captured the second season (weekly). I have pretty good results, however I am not able to induce holidays or specific dates. Maybe exog in SARIMAX that josef suggested. OR.... facebook launched a forecast library called prophet. It has the holidays and multiseasonality features. Give it a try.
https://facebook.github.io/prophet/docs/quick_start.html
